module_utils/network_lsr/argument_validator.py

```diff
@@ -596,6 +596,28 @@ def _validate_impl(self, value, name):
             )
         return Util.mac_ntoa(addr)
 
+class ArgValidatorClonedMac(ArgValidatorStr):
```

Comment:
instead of adding a new type, just extend (and use) `ArgValidatorMac`. Add an `enum_values` parameter to `ArgValidatorMac.__init__()` and set it to `enum_values=["preserve", "permanent", "random", "stable"]`.

But of course, this `enum_values` argument needs to be implemented by `ArgValidatorMac` itself, and *not* via `v = ArgValidatorStr._validate_impl(self, value, name, enum_values=enum_values)`.